### DISCUSSION TITLE
Would like to understand what's the reason for `@storybook/test` to have testing-library pkgs built-in

### DISCUSSION THREAD

**Discussion by csantos1113** [storybookjs/storybook#27528] [General]
I'm trying to move away from https://github.com/storybookjs/testing-library to https://github.com/storybookjs/storybook/blob/v7.6.19/code/lib/test (I'm not ready to go up to 8.x)

And I thought it was going to be just find and replace; but it doesn't seem to be the case. 

I've a lot of unit tests throwing `act()` warnings like:

<img width="785" alt="image" src="https://github.com/storybookjs/storybook/assets/87080313/f487a30f-3a54-48c4-8617-b0f8853fb9cd">

## Triage

- I have cloned https://github.com/storybookjs/storybook/blob/v7.6.19 
- I've built the packages locally
- and, after a lot of try and catch I narrowed the issue down to the following lines:

https://github.com/storybookjs/storybook/blob/0264a143c8de13978a841dca31c60c34666de363/code/lib/test/package.json#L71-L74

the `act()` warnings go away once I removed those 3 lines, built the package, and use that package.

### Why I think this is happening?

I believe this is because the version of `@testing-library/xxx` I have installed are different to the ones that are baked into `@storybook/test` dist folder.

## `noExternal` prop

None of other storybook packages use this feature of baking dependencies into it. **I wonder why `@storybook/test` is the exception?**

I see it was added in https://github.com/storybookjs/storybook/commit/f42e9efbba01ab40c5cdd770e6fc96aaf6ed189a by @kasperpeulen but I've no clue as why

**Note:** I see that list still exists in `next` https://github.com/storybookjs/storybook/blob/fdadc4114b0407c4dd3f6a2e30d10724d7ef5bf1/code/lib/test/package.json#L71-L74 - but I don't think it matters for my use case

---

Appreciate any help 🙏

**Comment by kasperpeulen** [comment]
cc @yannbf You also had this problem with act warnings right?

@csantos1113 We pre-bundle those packages for performance reasons. I will invrestigate what causes the act difference when prebundling.

**Reply by csantos1113** [reply]
Hey guys, I wonder if we should convert this to `issue`?

Thank you!

**Reply by kasperpeulen** [reply]
I think this issue is similar, and will be fixed in 8.2:
https://github.com/storybookjs/storybook/issues/27173

Also going to be patched back to 8.1.

BTW, upgrading to 8 doesn't require changes for lots of storybooks.
I think the main breaking change is if you use spies in play functions, which need to be explicit now.
